scalameter-core/src/main/scala/org/scalameter/utils/ClassPath.scala

```diff
@@ -8,51 +8,73 @@ import org.apache.commons.lang3.SystemUtils
 
 
 
-object ClassPath {
-  /** Returns the default classpath string.
+class ClassPath private (val paths: List[String]) extends Serializable {
+  /** Returns platform dependent classpath string.
    */
-  def default: String = {
-    extract(this.getClass.getClassLoader, sys.props("java.class.path"))
+  def mkString: String = {
+    val cp = paths.mkString(File.pathSeparator)
+    if (SystemUtils.IS_OS_WINDOWS) "\"" + cp + "\"" else cp
+  }
+
+  /** Prepends library to the classpath.
+   */
+  def ::(library: String) = {
+    ClassPath.validate(library)
+    new ClassPath(library :: paths)
+  }
+
+  override def equals(obj: Any): Boolean = obj match {
+    case cl: ClassPath => paths == cl.paths
+    case _ => false
+  }
+
+  override def hashCode(): Int = paths.hashCode()
+}
+
+object ClassPath {
+  private def validate(elem: String) = {
+    require(!elem.contains(File.pathSeparatorChar), s"Classpath library contains illegal character: ${File.pathSeparatorChar}")
+    if (SystemUtils.IS_OS_WINDOWS) require(!elem.contains("\""), "Classpath library contains illegal character: \"")
   }
 
-  /** Returns the default classpath string, surrounded by quotes on Windows.
+  private def fromString(classPath: String): ClassPath = {
+    val cl = if (SystemUtils.IS_OS_WINDOWS) classPath.stripPrefix("\"").stripSuffix("\"") else classPath
```

Comment:
Just for clarification: I should remove  both stripping quotes in `fromString` and adding them on `mkString`?